### ISSUE TITLE
attach'ed hbargraph is removed if first bus gets terminated

### ISSUE THREAD

**Issue by Jikstra** [grame-cncm/faust#785]
Example code: `process = _, (2.0 <: _, hbargraph("test", 0, 5) : attach) : _,!;`
Diagram: ![Screenshot_20220819_123440](https://user-images.githubusercontent.com/34889164/185600735-79d4c152-358e-4d44-bc7b-55ccd98e8b65.png)
 
 
 Expected: I would expect that an attached signal is getting exposed to the ui, even if it gets later on terminated
Actual: attached signal is getting removed, even if it's part of the diagram

 But not exposed in ui: ![Screenshot_20220819_123538](https://user-images.githubusercontent.com/34889164/185600862-04b0b59f-ae8f-46b2-9f47-3e4581d1f572.png)

**Comment by sletz** [comment]
The SVG diagram is a bit misleading here: in this cas the compiler actually remove the attached part and generates this C++ code:
```
virtual void compute(int count, FAUSTFLOAT** RESTRICT inputs, FAUSTFLOAT** RESTRICT outputs) {
        FAUSTFLOAT* input0 = inputs[0];
	FAUSTFLOAT* output0 = outputs[0];
	for (int i0 = 0; i0 < count; i0 = i0 + 1) {
		output0[i0] = FAUSTFLOAT(float(input0[i0]));
	}
}
```

**Comment by josmithiii** [comment]
Yes, `attach` must attach to a signal that _survives_.  It would be nice to have a primitive that works like you expected, e.g., `hbargraph("test", 0, 5) : persist`.

**Comment by Jikstra** [comment]
> Yes, `attach` must attach to a signal that _survives_. It would be nice to have a primitive that works like you expected, e.g., `hbargraph("test", 0, 5) : persist`.

Yes I think something like this would be really helpful. Would this be hard to implement?

**Comment by sletz** [comment]
Read the attach primitive documentation again: https://faustdoc.grame.fr/manual/syntax/#attach-primitive. Since your program finally compute an output, you should be able to use if in the attach expression right?

**Comment by Jikstra** [comment]
> Read the attach primitive documentation again: https://faustdoc.grame.fr/manual/syntax/#attach-primitive. Since your program finally compute an output, you should be able to use if in the attach expression right?

I'm currently writing a mixer with peak/dbmeter. It has 4 stereo buses for the channel, but they get terminated once they got mixxed down for the main output. So those signals don't survive, and also the attach primitive doesn't. I worked around this by keeping all the buses, as I write the interaction with the jack audio api myself, i can hide them there.

**Comment by sletz** [comment]
But you need to get the result of peak/dbmeter in a bargraph right? Why cant you attach them to the signal which is then mixe to main output ?
Any "simplest" piece of code that should the problem?

**Comment by Jikstra** [comment]
> But you need to get the result of peak/dbmeter in a bargraph right? Why cant you attach them to the signal which is then mixe to main output ? Any "simplest" piece of code that should the problem?

As I said, I guess I could make it work somehow, doing the metering before the mixdown. But my code would look something like this:
```
_,_ <: _,_,_,_  : _,_ (_,_:>_) : _, (_, (_metering : hbargraph(...)) : attach) : _,_
```

With a possible persist primtive
```
_,_ <: _,_,_,_  : _,_ (_,_:>_:metering:hbargraph(...):persist:!) : _,_
```

With persist i think it's more clear what happens. With attach it's why 2 inputs? What is the sense of feeding the right channel into it?

**Comment by nuchi** [comment]
Coincidentally, I just wrote a library to solve this problem: https://github.com/nuchi/faust-tap-library/

Actually I wanted something slightly different than a `persist` primitive — I wanted to route deeply nested signals to output without manually dealing with crossings, but when I saw this issue it was only a few extra lines to add `persist` functionality.

The name ("tap.lib") comes from tapping a delay line.

Please make sure to look at the "Limitations" section in the readme, there are some situations that it can't handle.

Example code:
```faust
tap = library("tap.lib");

A(k) = *(k);

complicatedExpression = _,_,_,_ : (
        (A(1) <: _,(hbargraph("first", 0, 1) : tap.T1)),
         A(2),
         A(3),
        (A(4) <: _,(hbargraph("second", 0, 1) : tap.T2)),
         A(5)
    ) ~ (_,_,_,_,_ :> _ <: _,tap.T3)
      : si.block(5)
      ;
process = tap.persist(
    tap.extract(complicatedExpression, tap.T3),
    (tap.T1, tap.T2)
);
```

Resulting block diagram:
![persist](https://user-images.githubusercontent.com/16771734/186065727-de18e9ef-0e42-4278-8832-dd5b2346d7b9.svg)

**Comment by josmithiii** [comment]
Nice!  I've often needed such taps
